esm/interface/css-style-declaration.js

```diff
@@ -13,9 +13,9 @@ const updateKeys = style => {
     if (attr) {
       style.set(PRIVATE, attr);
       for (const rule of attr[VALUE].split(/\s*;\s*/)) {
-        const pair = rule.split(/\s*:\s*/);
-        if (1 < pair.length) {
-          let [key, value] = pair;
+        const [key, ...rest] = rule.split(':');
+        const value = rest.join(':');
```

Comment:
do you mind moving this *after* the `if` check? 'cause it's a completely pointless operation if `rest.length` is `0`, right?

Reply:
on top of that, value should be `rest.join(':').trim()` within the `if` 😉 ... also, if you use `/\s*:\s*/` in the split the `trim` operation will be faster because the split already took care of dropping spaces, and I don't think there are cases with meant spaces around colons, including URLs that are badly formatted if that's the case

Reply:
Made that optimisation.

There are cases where spaces around a colon could be relevant like with `content`. Do you mind keeping it as is? My guess is that the RegExp creation might be just as expensive as trimming additional space on one end the string.

Reply:
The static RegExp can be easily optimized by engines, but fair enough about the content case, although the trim is still needed

Reply:
Thanks. Then I think this is ready ☺️

Do you need a test case for this?

Reply:
P.S. really nice codebase — used this for the first time this afternoon, hit the error and could find and fix it super easily.

Reply:
Ideally I’d like a test case as the code is covered but your use case is not double checked